### ISSUE TITLE
Proposal: zfs diff command support i18n.

### ISSUE THREAD

**Issue by patineboot** [openzfs/zfs#13550]
### Describe the feature would like to see added to OpenZFS

**The `zfs diff` command supports i18n.**
I have been faced with the zfs diff command printing the escaped characters.
e.g., the `zfs diff` command prints '\0343\0201\0202\0343\0201\0202\0343\0201\0202' escaped characters.

The cause is the `zfs diff` command missing the -h option.

### How will this feature improve OpenZFS?

We will get diff in all languages in the world.

### Additional context

**Problem Details:**

The `zfs diff` command is missing the `-h` option.
See OpenZFS Specification for the `-h` option on the `zfs diff` command:
<https://openzfs.github.io/openzfs-docs/man/8/zfs-diff.8.html>

1. Make the file in Japanese.
2. Get difference with the `zfs diff` command.

```bash
$ touch あああ
$ sudo zfs diff pool1@zfs-auto-snap_hourly-2022-06-12-0352 pool1

M	/pool1/
+	/pool1/\0343\0201\0202\0343\0201\0202\0343\0201\0202
```

**Software Environment:**

Ubuntu version: 22.04
OpenZFS version: 2.1.2

ZFS Pool Options

```bash
zpool create \
    -o ashift=12 -o autotrim=on \
    -O dnodesize=auto -O normalization=formD -O relatime=on \
    -O acltype=posixacl -O xattr=sa \
```

**Comment by rincebrain** [comment]
That's not a spec, that's just the man pages from git, where it's documented because it's been implemented and the man pages were updated at the same time. So it's already implemented and merged, just not in the stable releases at the moment.

It's even already included in the patches for 2.1.5, whenever that gets released.

In the interim, I personally just pipe through this alias in my shell for human-reading `zfs diff` output, though I make no promises that malicious inputs can't burn your terminal down, since it is interpreting the escaped characters:
`alias zfilt='while read -r LINE;do echo -e "${LINE}";done'`

**Comment by patineboot** [comment]
Thank you, @rincebrain.

I understood the site is the man pages are built from git.
I am waiting for the 2.1.5 or later version as the stable release.
